tests/std/tests/GH_002655_alternate_name_broke_linker/test.cpp

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <yvals.h>
+
+#if defined(_M_IX86) | defined(_M_X64)
+extern "C" int __declspec(dllimport) __stdcall __std_init_once_begin_initialize(void**, unsigned long, int*, void**);
+extern "C" int __declspec(dllimport) __stdcall __std_init_once_complete(void**, unsigned long, void*);

```

Comment:
`__declspec(dllimport)` doesn't make sense when we're linking statically. These should probably disappear when `_DLL` is not defined.

Reply:
because these are aliases for stuff from kernel32.dll they are "always" dllimport (the call needs to be generated with those semantics)

Reply:
See in <mutex> where we always use dllimport unless we're doing the clr workaround.

Reply:
Derp. Thanks for ignoring me =)